test/v3/initBuffer/initBuffer.integration.test.ts

```diff
@@ -0,0 +1,230 @@
+    parseUnits,
+    publicActions,
+    TestActions,
+    walletActions,
+} from 'viem';
+
+import {
+    Slippage,
+    Hex,
+    CHAINS,
+    ChainId,
+    InitBufferInput,
+    InitBufferV3,
+    Permit2Helper,
+    PERMIT2,
+    PublicWalletClient,
+    BALANCER_BUFFER_ROUTER,
+} from '../../../src';
+import {
+    TOKENS,
+    sendTransactionGetBalances,
+    setTokenBalance,
+    approveSpenderOnToken,
+    approveSpenderOnPermit2,
+} from '../../lib/utils';
+import { ANVIL_NETWORKS, startFork } from '../../anvil/anvil-global-setup';
+import { InitBufferBuildCallInput } from '@/entities/initBuffer/types';
+
+const chainId = ChainId.MAINNET;
+const USDC = TOKENS[chainId].USDC;
+const gearboxUSDC = TOKENS[chainId].gearboxUSDC;
+
+describe('Buffer Init', () => {
+    let client: PublicWalletClient & TestActions;
+    let rpcUrl: string;
+    let snapshot: Hex;
+    let testAddress: Address;
+    const initBuffer = new InitBufferV3();
+
+    beforeAll(async () => {
+        ({ rpcUrl } = await startFork(ANVIL_NETWORKS[ChainId[chainId]]));
+
+        client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+
+        const accounts = await client.getAddresses();
+
+        testAddress = accounts[0];
+
+        await setTokenBalance(
+            client,
+            testAddress,
+            USDC.address,
+            USDC.slot as number,
+            parseUnits('100000', USDC.decimals),
+        );
+
+        // Approve Permit2 to spend user tokens
+        await approveSpenderOnToken(
+            client,
+            testAddress,
+            USDC.address,
+            PERMIT2[chainId],
+        );
+
+        snapshot = await client.snapshot();
+    });
+
+    beforeEach(async () => {
+        await client.revert({
+            id: snapshot,
+        });
+        snapshot = await client.snapshot();
+    });
+
+    describe('permit 2 direct approval', () => {
+        beforeEach(async () => {
+            // Approve Buffer Router to spend tokens on the users behalf via Permit2
+            await approveSpenderOnPermit2(
+                client,
+                testAddress,
+                USDC.address,
+                BALANCER_BUFFER_ROUTER[chainId],
+            );
+        });
+
+        test('init', async () => {
+            const initBufferInput: InitBufferInput = {
+                chainId,
+                rpcUrl,
+                underlyingAmountIn: {
+                    rawAmount: parseUnits('100', USDC.decimals),
+                    decimals: USDC.decimals,
+                    address: USDC.address,
+                },
+                wrappedAmountIn: {
+                    rawAmount: parseUnits('0', gearboxUSDC.decimals),
+                    decimals: gearboxUSDC.decimals,
+                    address: gearboxUSDC.address,
+                },
+            };
```

Comment:
You might consider moving this input declaration to global scope and assign the value in beforeAll since is the same for both w/ and w/o permit2

```
let initBufferInput: InitBufferInput;
...

        initBufferInput = {
            chainId,
            rpcUrl,
            underlyingAmountIn: {
                rawAmount: parseUnits('100', USDC.decimals),
                decimals: USDC.decimals,
                address: USDC.address,
            },
            wrappedAmountIn: {
                rawAmount: parseUnits('0', gearboxUSDC.decimals),
                decimals: gearboxUSDC.decimals,
                address: gearboxUSDC.address,
            },
        };
```

Reply:
Also just to make sure, the reason you're setting wrappedAmountIn to 0 is because would be a pain to do the token slot thingy to set test user balance for gearbox? 

But if user did have wrapped amount they could use them as part of init right?

Reply:
Yeah, any amount of wrapped/underlying can be provided to init the buffer. I set it to zero to avoid the setup pain 😅 
You could think that since it's zero, we could set the token to undefined, but we need the address to be provided.